openedx/openedx-platform
# Auto-advance video toggle

This PR is based on https://github.com/edx/edx-platform/pull/15509

# Description

This adds an auto-advance toggle to the video player. If turned on, as soon as the video ends the next unit or subsection will be loaded.

Since this is a global preference, when it is set loading a section with a video will automatically start it. This has the (desired) effect that consecutive video sections will play automatically from start to finish.

A course level setting is also provided to enable video auto advance by default.

A platform level setting (feature flag) is also provided as a fallback value that enables to enable video advance globally. Video advancing is disabled by default.

# Dependencies

N/A

# JIRA

https://openedx.atlassian.net/browse/OSPR-1874

# edX Sandbox URL
https://studio-video-auto-advance.sandbox.edx.org/container/block-v1:edX+DemoX+Demo_Course+type@vertical+block@vertical_0270f6de40fc

# Sandbox URLs

- LMS: https://pr15803.sandbox.opencraft.hosting/courses/course-v1:edX+DemoX+Demo_Course/courseware/d8a6192ade314473a78242dfeedfbf5b/edx_introduction/
- Studio: https://studio-pr15803.sandbox.opencraft.hosting/

# Testing instructions

1. In Studio add a new subsection with two units with video components and publish them.
1. In the LMS go to the first unit of the new subsection, play the video and enable the "Auto-advance" button on the video (it should turn blue).
1. Wait for the video to end. The next unit should load automatically, and the video should start playing with no user input.
1. Go back to the first unit of the new subsection and disable the "Auto-advance" button.
1. Wait for the video to end. The next unit should not load automatically.
1. In Studio go to the Advance Settings page of the course and set "Enable video auto-advance" to true.
1. In the LMS go to (or reload) the first unit of the new subsection and the "Auto-advance" button should be enabled.
1. Wait for the video to end. The next unit should load automatically, and the video should start playing with no user input.
1. Disable "Enable video auto-advance" for the course
1. Disable also auto-advance for the current video
1. In lms.env.json, add "AUTOADVANCE_VIDEOS": true to the FEATURES section
1. Restart. All videos should have auto-advance enabled
